### PR TITLE
TrackData: Group all sequence-bound properties

### PR DESCRIPTION
**Pull request by Gymnasiast**
This massively refactors how track element sequences are handled, grouping them together in a struct of their own and saving their size, rather than having to search for the end marker manually. This results in more legible code, but also makes it much easier to refactor the track drawing code, which is why I embarked on it in the first place.

### CODE REVIEW

File: src/openrct2/actions/TrackPlaceAction.cpp [hunk 1]
```diff
@@ -509,7 +509,8 @@ GameActions::Result TrackPlaceAction::Execute() const
             else
             {
                 // Remove walls in the directions this track intersects
-                uint8_t intersectingDirections = wallEdges[blockIndex];
+                uint8_t intersectingDirections = ted.sequences[blockIndex].allowedWallEdges;
+                ;
```
Comment (AaronVanGeffen) [inline src/openrct2/actions/TrackPlaceAction.cpp:513]: Stray semicolon?

**Review by AaronVanGeffen — COMMENTED** [review]
We now have a `WoodenSupports.h` and `WoodenSupports.hpp`. This is bound to create confusion, so I think they should be merged.

**Review by AaronVanGeffen — COMMENTED** [review]
The new format for `TrackData.cpp` is much more readable, so that's a big win. (I hope these changes won't conflict too badly with any new track styles currently in development, though...)

**Comment by Gymnasiast** [comment]
> We now have a `WoodenSupports.h` and `WoodenSupports.hpp`. This is bound to create confusion, so I think they should be merged.

Those were not introduced in this PR, though.

**Comment by ZeeMaji** [comment]
The go karts seem to be using the wrong supports on their turn pieces (should use full truss pieces rather than corner truss pieces)

0.4.14 release (also checked develop and it's the same there)
Screenshot at 2024-09-16 06-52-22
This PR
Screenshot at 2024-09-16 06-51-40

**Comment by AaronVanGeffen** [comment]
> > We now have a `WoodenSupports.h` and `WoodenSupports.hpp`. This is bound to create confusion, so I think they should be merged.
> 
> Those were not introduced in this PR, though.

Ah, I didn't realise. In that case, let's merge them in a separate PR.

**Comment by ZeeMaji** [comment]
The supports on the classic wooden coaster's small banked turn are broken.
0.4.14:
Screenshot at 2024-09-19 10-15-01
This PR:
Screenshot at 2024-09-19 10-14-55

**Review by ZeeMaji — APPROVED** [review]
After inspecting a few proper parks as well as building ten different ride types from scratch, everything seems to be working as expected.

File: src/openrct2/ride/TrackData.h [hunk 2]
```diff
@@ -81,13 +82,30 @@ namespace OpenRCT2::TrackMetaData
...
 
+    struct SequenceMetalSupport
+    {
+        MetalSupportPlace place = MetalSupportPlace::None;
+        uint8_t alternates = false;
+    };
+
+    struct SequenceDescriptor
+    {
+        SequenceClearance clearance{};
```
Comment (spacek531) [inline src/openrct2/ride/TrackData.h:93]: It appears these properties could be inlined into Sequence Descriptor. Something for a followup PR if true.

**Comment by spacek531** [comment]
Mini coaster testing:

RightUp25Curve5Tiles sequence 6
image

LeftDown25Curve5Tiles sequence 0
image

Left Helix Up Small sequence 3
image

Right Helix Down Small sequence 4
image

**Comment by spacek531** [comment]
Tested these and they all passed visual inspection:
- wooden roller coaster
- wooden wild mouse
- corkscrew coaster
- inverted coaster
- inverted monorail

**Comment by Gymnasiast** [comment]
@spacek531 I didn’t touch a single steel support in this PR (I just added a currently-unused table), so I suggest you check that issue with the develop branch.

**Comment by spacek531** [comment]
This was tested by overlaying giant screenshots of the track zoo from develop and feature branches. The screenshots in the post are of the feature branch.On Sep 19, 2024, at 22:54, Michael Steenbeek ***@***.***> wrote:﻿
@spacek531 I didn’t touch a single steel support in this PR (I just added a currently-unused table), so I suggest you check that issue with the develop branch.

File: src/openrct2/ride/TrackData.h (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/TrackData.h:93]: Suggestions for a better name are welcomed.

**Comment by Gymnasiast** [comment]
I checked it on myself, but could not reproduce the issue on either this branch or develop. I’m quite certain this PR cannot cause it as it doesn’t touch anything metal or Mini RC-related.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**